apps/openmw/mwworld/cellpreloader.cpp

```diff
@@ -438,28 +443,14 @@ namespace MWWorld
         }
     }
 
-    void CellPreloader::abortTerrainPreloadExcept(const CellPreloader::PositionCellGrid *exceptPos)
-    {
-        const float resetThreshold = ESM::Land::REAL_SIZE;
-        for (const auto& pos : mTerrainPreloadPositions)
-            if (exceptPos && (pos.first-exceptPos->first).length2() < resetThreshold*resetThreshold && pos.second == exceptPos->second)
-                return;
-        if (mTerrainPreloadItem && !mTerrainPreloadItem->isDone())
-        {
-            mTerrainPreloadItem->abort();
-            mTerrainPreloadItem->waitTillDone();
-        }
-        setTerrainPreloadPositions(std::vector<CellPreloader::PositionCellGrid>());
-    }
-
-    bool contains(const std::vector<CellPreloader::PositionCellGrid>& container, const std::vector<CellPreloader::PositionCellGrid>& contained)
+    bool contains(const std::vector<CellPreloader::PositionCellGrid>& container, const std::vector<CellPreloader::PositionCellGrid>& contained, float tolerance=1.f)
```

Comment:
Put not declared in the header free function into anonymous namespace to avoid possible unintended symbol usage from other translation units https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed2 .

`const std::vector<CellPreloader::PositionCellGrid>& contained` can be replaced by a template argument to avoid unnecessary `std::vector` allocation for a single value. I'd suggest to use `std::span`, but OpenMW doesn't support C++20. `std::array` or `std::initializer_list` can be used on a caller side for a single value container:
```c++
template <class Contained>
bool contains(const std::vector<CellPreloader::PositionCellGrid>& container,
              const Contained& contained, float tolerance=1.f) { /*...*/ }

contains(mTerrainPreloadPositions, std::array {*exceptPos}, ESM::Land::REAL_SIZE);
```
CTAD allows to write `std::array {*exceptPos}` without specifying type and size.